sdk/tables/azure-data-tables/src/table_clients.cpp

```diff
@@ -1062,10 +902,11 @@ std::string TableClient::PrepMergeEntity(std::string const& changesetId, Models:
   std::string returnValue = "--" + changesetId + "\n";
   returnValue += "Content-Type: application/http\n";
   returnValue += "Content-Transfer-Encoding: binary\n\n";
-
-  returnValue += "MERGE " + m_url.GetAbsoluteUrl() + "/" + m_tableName + PartitionKeyFragment
-      + entity.GetPartitionKey().Value + RowKeyFragment + entity.GetRowKey().Value + ClosingFragment
-      + " HTTP/1.1\n";
+  auto url = m_url;
+  url.AppendPath(
+      m_tableName + PartitionKeyFragment + entity.GetPartitionKey().Value + RowKeyFragment
+      + entity.GetRowKey().Value + ClosingFragment);
+  returnValue += "MERGE " + url.GetAbsoluteUrl() + " HTTP/1.1\n";
```

Comment:
Manually building the HTTP headers is a rather non-standard way of making requests, typically this is done by creating an Azure::Core::Http::Request object and adding the headers to those.

Is there a reason for the alternate mechanism?

Reply:
This is not a request per se, it's a body of request  that contains a series of requests , in a certain magical format .

Reply:
E.G.  
  
--batch_a1e9d677-b28b-435e-a89e-87e6a768a431  
Content-Type: multipart/mixed; boundary=changeset_8a28b620-b4bb-458c-a177-0959fb14c977  
  
--changeset_8a28b620-b4bb-458c-a177-0959fb14c977  
Content-Type: application/http  
Content-Transfer-Encoding: binary  
  
POST https://myaccount.table.core.windows.net/Blogs HTTP/1.1  
Content-Type: application/json  
Accept: application/json;odata=minimalmetadata  
Prefer: return-no-content  
DataServiceVersion: 3.0;  
  
{"PartitionKey":"Channel_19", "RowKey":"1", "Rating":9, "Text":".NET..."}  
--changeset_8a28b620-b4bb-458c-a177-0959fb14c977  
Content-Type: application/http  
Content-Transfer-Encoding: binary  
  
POST https://myaccount.table.core.windows.net/Blogs HTTP/1.1  
Content-Type: application/json  
Accept: application/json;odata=minimalmetadata  
Prefer: return-no-content  
DataServiceVersion: 3.0;  
  
{"PartitionKey":"Channel_17", "RowKey":"2", "Rating":9, "Text":"Azure..."}  
--changeset_8a28b620-b4bb-458c-a177-0959fb14c977  
Content-Type: application/http  
Content-Transfer-Encoding: binary  
  
 
--changeset_8a28b620-b4bb-458c-a177-0959fb14c977--  
--batch_a1e9d677-b28b-435e-a89e-87e6a768a431

Reply:
@LarryOsterman what do you think about adding support for such Request creation scenarios for service clients within Azure::Core::Http, including `multipart/mixed`? I assume this isn't unique to tables only, and other services could benefit from it.

I worry about us having bespoke string creation for the request, if it can benefit from a common helper type in Core, which we could test in isolation, and use by multiple service SDKs.

If this is only expected to be used by tables, or has a format that's very specific to that service, then we can ignore this concern.

Btw, I noticed some url encoding issue, at a quick glance, we should consider vetting this approach and double check the data flow: https://github.com/Azure/azure-sdk-for-cpp/issues/6225

Reply:
Based on George's description, I'm firmly against it, *unless* we discover that other Azure Services require support for the multipart/mixed MIME type. 

The rule we have held up until now is that we don't add functionality to Azure::Core *unless* that functionality is required by two or more packages. As far as I know, the use of multipart/mixed is unique to the Tables client 

To my knowledge, none of the other SDKs have Azure Core support for creating multipart/mixed requests. If we do discover multiple service clients which require multipart/mixed support, we can easily retrofit this package to support whatever we come up with since all of this is 100% internal.